### ISSUE TITLE
Question on slashing module

### ISSUE THREAD

**Issue by entergmode** [cosmos/cosmjs#1259]
Hi,

As i see the slashing module is not imported to stargateclient.ts
Not sure how we can use it to query? Maybe i'm missing something

Appreciate the answer. Thanks!!

**Comment by entergmode** [comment]
Hi, @webmaster128 , not sure my question is clear. I meant `https://github.com/cosmos/cosmjs/blob/main/packages/stargate/src/stargateclient.ts` doesnt import slashing module to export it. How do we use this slashing module then? thanks

**Comment by webmaster128** [comment]
Yeah, so there is no great answer here, bu maybe I can help. The StargateClient/SigningStargateClient only install the query extensions they need to power the higher level methods that are implemented there. They do not include slashing. Their query client is private and should not be used directly. If you need a `QueryClient` with custom extensions, you can create one on you own, like here: https://github.com/cosmos/cosmjs/blob/v0.29.0-rc.1/packages/stargate/src/modules/slashing/queries.spec.ts#L12

This is the query part. For slashing, no messages are implemented. Do they exist?
